python/tests/test_nessie_cli.py

```diff
@@ -315,7 +316,36 @@ def test_merge() -> None:
     # Passing detached commit-id plus a _different_ hash-on-ref --> error
     execute_cli_command(["merge", f"dev@{dev_hash}", "-c", main_hash, "-o", main_hash], ret_val=1)
 
-    execute_cli_command(["merge", "dev", "-c", main_hash])
+    assert_that(execute_cli_command(["merge", "dev", "-c", main_hash])).starts_with("The following 1 commits were merged onto main:")
+    logs = simplejson.loads(execute_cli_command(["--json", "log"]))
+    # we don't check for equality of hashes here because a merge
+    # produces a different commit hash on the target branch
```

Comment:
could we maybe verify things like the commit msg still being present or the like?

Reply:
nit: if the log cmd constains the commit msg maybe verify them as well here